### PR TITLE
feat(docs): add documentation site and GitHub Pages deployment

### LINKED ISSUE #973: feat(docs): add documentation site and GitHub Pages deployment
## Summary

Add a public documentation site for py-lintro, deployed to GitHub Pages at `https://lgtm-hq.github.io/py-lintro/`. The site should include a marketing-style home page (like Rustume and turbo-themes), render the existing `docs/` markdown content, and bundle the Python coverage HTML report alongside the site. Use the Rustume documentation-site stack and lgtm-ci workflows as the starting template.

This is **infrastructure and content publishing** — separate from [#479](https://github.com/lgtm-hq/py-lintro/issues/479) (`lintro docs` polyglot generation CLI), which remains a future effort.

## Problem

Today, py-lintro documentation lives only as raw markdown in the `docs/` folder, linked from the README via GitHub's file viewer. There is no browsable, searchable documentation site, no consistent branding, and no single entry point for users.

GitHub Pages currently publishes **coverage reports only** (via `test-ci.yml` → lgtm-ci `reusable-test-python-publish.yml`). Coverage and docs are not unified under one URL, and the project lacks the polished public presence that sibling repos (Rustume, turbo-themes) already have.

## Proposed Solution

Adopt **Rustume's Model B** (site + bundled reports) as the baseline architecture:

```
https://lgtm-hq.github.io/py-lintro/           → Astro documentation site + home page
https://lgtm-hq.github.io/py-lintro/coverage/  → Python pytest HTML coverage report
```

### Site structure

```
apps/site/
├── src/
│   ├── pages/
│   │   ├── index.astro              # Marketing home page
│   │   └── docs/[...slug].astro     # Docs content pages
│   ├── content/docs/                # Migrated from repo-root docs/
│   ├── layouts/                     # BaseLayout, DocsLayout (adapt from Rustume)
│   ├── components/                  # Header, Footer, sidebar, search, etc.
│   └── data/                        # docs-nav.ts, site-links.ts, landing.ts
├── public/                          # Static assets (logo, favicon)
├── astro.config.mjs
├── package.json
└── bun.lock

scripts/ci/site/
├── build.sh                         # ASTRO_BASE from defaults.env
├── check.sh                         # astro check
├── test.sh                          # Vitest
├── preview-serve.sh
├── preview-pages-local.sh
├── defaults.env                     # ASTRO_BASE_DEFAULT=/py-lintro/
└── README.md
```

### CI/CD workflows

| Workflow | Purpose |
| -------- | ------- |
| `site-quality.yml` | PR/main: build site, lychee link check, vitest |
| `deploy-pages.yml` | Deploy via lgtm-ci `reusable-deploy-site-with-reports` |

`deploy-pages.yml` triggers on `workflow_run` after **Quality - Documentation Site** (or the main test workflow that uploads coverage) succeeds on `main`, plus `workflow_dispatch`.

### Pages bundle manifest

```json
{
  "strict": false,
  "bundles": [
    {
      "id": "python-coverage",
      "workflow": "Test - CI Pipeline",
      "artifact": "coverage-report-html",
      "dest": "coverage",
      "require_success": true
    }
  ]
}
```

(Exact workflow/artifact names should match py-lintro's current CI outputs.)

### Tech stack (match Rustume)

- **Astro 6** static site
- **Bun** package manager
- **@lgtm-hq/turbo-themes** for theming
- **Pagefind** for client-side search
- **Vitest** for site unit tests
- **lgtm-ci** reusables for hardened CI (`setup-node`, `harden-runner`, `reusable-deploy-site-with-reports`)

### Content migration

Migrate existing `docs/` markdown into `apps/site/src/content/docs/`, preserving the current information architecture:

- Getting Started, Configuration, AI Features, Docker, GitHub Integration
- Tool Analysis pages
- Architecture (Vision, Roadmap, ARCHITECTURE)
- Contributing, Troubleshooting, Security

Keep repo-root `docs/` as the source of truth initially (copy or symlink strategy TBD in implementation), or move content into `apps/site/src/content/docs/` and update README links to point at the published site.

### Home page

Marketing-style landing page (like Rustume/turbo-themes), not docs-only:

- Hero with Lintro value proposition
- Feature highlights (unified CLI, 25+ tools, Docker, AI features)
- Quick-start CTA linking to `/docs/getting-started/`
- Links to GitHub, PyPI, coverage report

Visual design can start minimal and iterate — the priority is working infrastructure and content, not final branding.

### Local development

```bash
make site-dev       # http://localhost:4321/py-lintro/
make site-build
make site-preview
```

### Reference implementations

- **Primary:** [lgtm-hq/Rustume](https://github.com/lgtm-hq/Rustume) — `apps/site/`, `scripts/ci/site/`, `site-quality.yml`, `deploy-pages.yml`, `.github/pages-bundle-manifest.json`
- **Secondary:** [lgtm-hq/turbo-themes](https://github.com/lgtm-hq/turbo-themes) — older inline CI, but proven Astro + turbo-themes patterns

### Key files to create/modify

- `apps/site/` — new Astro site (scaffold from Rustume, adapt branding)
- `scripts/ci/site/` — build/check/test/preview scripts
- `.github/workflows/site-quality.yml` — site CI
- `.github/workflows/deploy-pages.yml` — Pages deploy via lgtm-ci
- `.github/pages-bundle-manifest.json` — coverage bundle config
- `Makefile` — add `site-dev`, `site-build`, `site-test`, `site-preview` targets
- `README.md` — add docs site link badge/URL
- `test-ci.yml` — ensure coverage HTML artifact name matches bundle manifest

### Prerequisites

- GitHub Pages source set to **GitHub Actions** (not branch deploy)
- lgtm-ci pin aligned with current py-lintro workflows (check `test-ci.yml` for current ref)
- Node ≥ 22, Bun (already used in repo)

### Potential challenges

1. **Coverage artifact naming** — current `reusable-test-python-publish.yml` may publish coverage differently than Rustume's separate Coverage Reports workflow; bundle manifest must match actual artifact names.
2. **Markdown migration** — `docs/` has cross-links and markdownlint config; rehype plugins (like Rustume's `rehype-doc-links.mjs`) may be needed to rewrite relative links for the site base path.
3. **Dual doc locations** — decide whether repo-root `docs/` remains canonical or content moves entirely into `apps/site/src/content/docs/`.
4. **Existing coverage-only Pages** — deploying the full site replaces the current coverage-only root; ensure badge URLs and codecov links still work.
5. **astro-check in monorepo** — `apps/site/` will add another Astro project; ensure lintro's astro-check tool config handles it (related: #940).

### Out of scope (this issue)

- Custom domain (`lintro.dev` etc.)
- `lintro docs` CLI command ([#479](https://github.com/lgtm-hq/py-lintro/issues/479))
- `lintro serve` web dashboard ([#512](https://github.com/lgtm-hq/py-lintro/issues/512))
- API reference generation from Python source (pdoc/mkdocstrings)
- Final visual design / branding polish

## Benefits

- **Discoverability** — searchable, navigable docs with home page and consistent UX
- **Professional presence** — matches the quality bar of Rustume and turbo-themes
- **Unified Pages URL** — docs and coverage under one domain (`lgtm-hq.github.io/py-lintro/`)
- **CI confidence** — link checking and site tests catch broken docs before merge
- **Foundation for growth** — infrastructure ready for future content without re-architecting

## References

- Rustume site CI: `scripts/ci/site/README.md`
- Rustume deploy: `.github/workflows/deploy-pages.yml`
- Current coverage Pages: `.github/workflows/test-ci.yml` (`publish-coverage` job)
- Existing docs hub: `docs/README.md`
- Related (separate): [#479](https://github.com/lgtm-hq/py-lintro/issues/479), [#512](https://github.com/lgtm-hq/py-lintro/issues/512)

### PR DESCRIPTION
**Pull request by TurboCoder13**
## Commit Summary (Conventional Commits)

- Title (required, present tense):

  ```text
  feat(docs): add documentation site and GitHub Pages deployment
  ```

- Type:
  - [x] feat (minor)
  - [ ] fix / perf (patch)
  - [ ] docs
  - [ ] refactor
  - [ ] test
  - [ ] chore / ci / style

- Breaking change:
  - [ ] `!` in title or `BREAKING CHANGE:` footer included

### Release Trigger Rules (exact)

- A merged PR will bump the version based on its title (squash merge required):
  - `feat(...)` or `feat:` → MINOR bump
  - `fix(...)` / `fix:` or `perf(...)` / `perf:` → PATCH bump
  - Any title with `!` after the type (e.g. `feat!:` or `feat(scope)!:`) or a body
    containing `BREAKING CHANGE:` → MAJOR bump
- Use squash merge so the PR title becomes the merge commit title.
- Valid examples:
  - `feat(cli): add --group-by`
  - `fix(parser): handle empty config`
  - `perf: optimize grouping performance`
  - `feat(api)!: remove deprecated flags`

## What's Changing

Adds a public Astro documentation site at `apps/site/`, deployed to GitHub Pages at
`https://lgtm-hq.github.io/py-lintro/` with bundled Python coverage HTML. Migrates
existing `docs/` markdown into site content, introduces site-quality CI (build, lychee,
Vitest), and deploy-pages workflow via lgtm-ci reusable actions.

Draft PR — ready for review of infrastructure and content; post-merge steps include
enabling GitHub Pages from Actions and updating the repo About URL.

## Checklist

- [x] Title follows Conventional Commits
- [x] Tests added/updated
- [x] Docs updated if user-facing
- [x] Local CI passed (`./scripts/local/run-tests.sh`)

## Closes

- Closes #973

### Site (`apps/site/`)

- Astro 6 + Bun static site with Terminal Brutalist default theme
- Migrated docs content with section tabs, dual-rail sidebar, on-page TOC, and Resources
  footer with SmartLink pills
- Home page, design gallery pages, Pagefind search, turbo-themes integration
- 103 Vitest tests with ~81% statement coverage on site lib/data modules

### CI / deployment

- `site-quality.yml` — migrate docs, build, lychee link check, Vitest + pytest
- `deploy-pages.yml` — lgtm-ci `reusable-deploy-site-with-reports` on main
- `.github/pages-bundle-manifest.json` — bundles `coverage-report-html` at `/coverage/`
- `test-ci.yml` — stages flat HTML coverage via `stage-python-coverage-html.sh`

### Local development

```bash
make site-dev       # http://127.0.0.1:4321/py-lintro/
make site-build
make site-test
make site-preview
```

### Testing strategy

- `uv run lintro chk` — zero issues
- `uv run lintro tst` — 5171 tests passed
- `make site-test` — 103 Vitest tests passed

### ⚠️ Maintainer action required before deploys go live (#1106)

The repo's Pages configuration is in a mixed state: `build_type` is already
`workflow` (GitHub Actions), but a leftover **branch source** (`main` / `/`)
is still set. While this mixed config persists, `deploy-pages.yml` can build
green yet the site will not reliably go live.

Before merging (admin-only, cannot be fixed from CI or the API safely):

> **Settings → Pages → Build and deployment → Source → "GitHub Actions"**

This clears the leftover branch source so Actions-based deploys stick.

### Docs link integrity (lychee) — resolved

The first-ever run of the `Check links` step surfaced 133 broken internal
links (pre-existing; the step had always been skipped by earlier failures).
Both categories are now fixed:

1. **`/coverage/`** — injected at deploy time by `deploy-pages.yml` via
   `pages-bundle-manifest.json`; the standalone `site-quality` build
   legitimately lacks it, so `prepare-lychee-action-args.sh` excludes it
   (checking it there tests the wrong artifact — documented in the script).
2. **`.md` cross-links** — `migrate-docs-content.py` now emits a generated
   source→route map (`src/generated/docs-route-map.ts`); the site's link
   layer (`doc-route-map.ts` + `doc-resources.ts`) resolves authored `.md`
   and directory links to final `/docs/<category>/<slug>/` routes, drops
   links to unmigrated docs, and rewrites repo-root README links to GitHub.
   Local lychee over the built dist: **0 errors**.

### Cleanup bundle (owner-approved)

- Docs content is now generated at build time from `docs/` (single source of
  truth): `src/content/docs/` is untracked and `bun run migrate` is chained
  into the dev/build/check/test package scripts.
- Removed ~35MB of unreferenced foreign (Rustume) assets under
  `public/assets/images/` and `public/assets/templates/`, plus the legacy
  thumbnail generator script.
- Removed the five standalone design-direction mockup pages and their gallery
  index (git history preserves them); the docs-navigation mockups remain.

> [!NOTE]
> ### Add Astro documentation site with GitHub Pages deployment for py-lintro
> - Adds a full [Astro site](https://github.com/lgtm-hq/py-lintro/pull/974/files#diff-231332a8b85c36daf1b5166bff6faa5e3abc16ab9842c4eadf9db6a2275cff05) under `apps/site/` with a homepage, docs hub, per-doc static routes, design mockup gallery, and a Pagefind-powered search dialog
> - Populates the docs collection with content covering getting started, tools (20+ analyzers), architecture, contributing, security, usage guides (configuration, Docker, GitHub integration, AI features, plugins, debugging, troubleshooting)
> - Adds `docs-nav`, `sidebar-nav`, `site-links`, `themes`, and `landing` TypeScript data modules with Vitest coverage to drive navigation, sidebar grouping, and theme selection
> - Introduces Makefile targets (`site-dev`, `site-build`, `site-test`, `site-preview`) and CI scripts under [scripts/ci/site/](https://github.com/lgtm-hq/py-lintro/pull/974/files#diff-ed2a5ebfbcdbc3d5dc8200c6086076553a0fa8779a00636e7e58756d137657b9) for building, type-checking, testing, link-checking, and local preview
> - Adds two GitHub Actions workflows: [site-quality.yml](https://github.com/lgtm-hq/py-lintro/pull/974/files#diff-20aacf48b3d575b0f16cb6c3d90b8dd9f763b81b0a29e1f24840893f8c620aec) builds and link-checks on push/PR, and [deploy-pages.yml](https://github.com/lgtm-hq/py-lintro/pull/974/files#diff-0f132b0962009071404990a6789b3466dbd8723d1f7b744f6aca0312396a7900) deploys to GitHub Pages on success of upstream CI workflows
> - Replaces the direct Pages coverage publish with a staging step that uploads a `coverage-html` artifact for bundling into the Pages deploy via [pages-bundle-manifest.json](https://github.com/lgtm-hq/py-lintro/pull/974/files#diff-e4ad31f14a15ee2a73b47f56e6a603a75e7ee3668309678009e4b25f42c5950f)
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized df3141f.</sup>
> 
>

## Summary by CodeRabbit

* **New Features**
  * Added a new documentation site with navigation, search, theme switching, docs pages, and homepage updates.
  * Introduced coverage and Pages deployment workflows for publishing site assets and reports.
  * Added utility scripts and helpers for building, testing, previewing, and migrating docs content.

* **Bug Fixes**
  * Improved link handling, URL normalization, and safe external-link behavior across site content.
  * Tightened Markdown and HTML sanitization for documentation pages.

* **Tests**
  * Added broader automated test coverage for docs navigation, routing, theming, links, and site content validation.

<details><summary><h3>Greptile Summary</h3></summary>

This PR adds a public documentation site and Pages deployment. The main changes are:

- New Astro site under `apps/site/`.
- Migrated docs content and generated route mapping.
- Pagefind search and site navigation components.
- GitHub Actions workflows for site quality checks and Pages deploys.
- Coverage HTML staging for the Pages bundle.
</details>

<h3>Confidence Score: 5/5</h3>

This looks safe to merge.

- No blocking issues found in the changed code.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| scripts/ci/testing/stage-python-coverage-html.sh | Stages coverage HTML from raw coverage data, prebuilt HTML, or coverage JSON. |
| scripts/ci/testing/render-coverage-json-html.py | Adds a small HTML renderer for coverage JSON fallback output. |
| .github/workflows/test-ci.yml | Uploads staged coverage HTML as an artifact for the Pages deploy bundle. |
| apps/site/package.json | Defines site scripts for migration, build, tests, type checks, and Pagefind indexing. |

</details>

<sub>Reviews (4): Last reviewed commit: ["fix(docs): document coverage JSON HTML h..."](https://github.com/lgtm-hq/py-lintro/commit/418039d5caa4edbeb7ba96465afc69b3a1e720c6) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=42343433)</sub>

### CODE REVIEW

File: apps/site/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  "version": "0.0.1",
+  "private": true,
+  "license": "MIT",
+  "engines": {
+    "node": ">=22.12.0"
+  },
+  "scripts": {
+    "migrate": "python3 ../../scripts/ci/site/migrate-docs-content.py",
+    "dev": "bun run migrate && astro dev",
+    "build": "bun run migrate && node scripts/copy-assets.mjs && astro build && bun x pagefind --site dist --glob '**/*.html' --exclude-selectors '[data-pagefind-ignore]' --exclude-selectors 'nav' --exclude-selectors 'footer'",
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/package.json:13]: Verified as a false positive. Astro's `base` (`/py-lintro/`) is a URL prefix, not an output-directory prefix: the build still writes to `apps/site/dist/` (default `outDir`), and Pagefind writes to `dist/pagefind/`. Since the deploy uploads `apps/site/dist` as the Pages artifact root — served at `https://lgtm-hq.github.io/py-lintro/` — `dist/pagefind/pagefind-ui.js` resolves to `/py-lintro/pagefind/pagefind-ui.js`, which is exactly what the generated HTML loads. No base mismatch. Resolving.

File: scripts/ci/testing/stage-python-coverage-html.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+if [[ ! -f "${COVERAGE_XML}" ]]; then
+	echo "::error::Missing coverage.xml at ${COVERAGE_XML}"
+	exit 1
+fi
+
+rm -rf "${OUTPUT_DIR}"
+mkdir -p "${OUTPUT_DIR}"
+
+uv run coverage html -d "${OUTPUT_DIR}" --data-file "${COVERAGE_XML}"
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply scripts/ci/testing/stage-python-coverage-html.sh]: Confirmed as a genuine issue (reproduced locally: `coverage html --data-file coverage.xml` fails with 'file is not a database'; it requires the binary `.coverage` data file). Additionally, the `python-coverage` artifact produced by the reusable test workflow contains `coverage.json` + `pytest-results.json` — neither `coverage.xml` nor a `.coverage` data file — so this main-only staging job cannot currently produce HTML from what it downloads. Important scoping notes: (1) this job is gated to `main` push and is skipped on PRs / in the merge queue, so it does not block this merge; (2) the Pages deploy tolerates a missing coverage bundle — the bundler runs with `strict:false`, so a failed/absent `coverage-html` bundle is warned-and-skipped (not fatal), and the deploy is additionally triggered by the Quality-Documentation-Site workflow independently — so the site still deploys. A correct fix requires plumbing the binary `.coverage` file (or a pre-rendered HTML report) through the reusable lgtm-ci test workflow's uploaded artifact, which is out of scope for this PR's inline diff. Tracking as a follow-up; resolving this thread as non-merge-blocking.

File: apps/site/scripts/copy-assets.mjs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+  }
+}
+
+if (existsSync(turboCssRoot)) {
+  copyDir(turboCssRoot, join(publicAssets, 'css'));
+  console.log('Copied turbo-themes CSS');
+} else {
+  throw new Error('turbo-themes CSS not found — run bun install in apps/site first');
+}
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/scripts/copy-assets.mjs:51]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/components/docs/DocsSectionNav.astro [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+interface Props {
+  category: DocCategory;
+  currentSlug: string;
+}
+
+const { category, currentSlug } = Astro.props;
+const base = import.meta.env.BASE_URL;
+
+const sectionDocs = (await getCollection("docs", ({ data }) => !data.draft && data.category === category))
+  .sort((a, b) => a.data.order - b.data.order);
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/components/docs/DocsSectionNav.astro:15]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/components/docs/DocsSidebar.astro [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    </select>
+  </div>
+  <div
+    class="sidebar-resize-handle"
+    id="docs-sidebar-resize"
+    role="separator"
+    aria-orientation="vertical"
+    aria-label="Resize sidebar"
+    tabindex="0"
+  ></div>
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/components/docs/DocsSidebar.astro:125]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/components/docs/DocsSidebar.astro [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+  }
+
+  function initSidebarChrome() {
+    if (!layout) {
+      return;
+    }
+    const collapsed = localStorage.getItem(COLLAPSED_KEY) === "true";
+    applyWidth(readWidth());
+    setCollapsed(collapsed);
+  }
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/components/docs/DocsSidebar.astro:193]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/components/Footer.astro [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  .footer-tagline { color: var(--turbo-text-muted, var(--turbo-text-secondary)); margin-top: 0.5rem; font-size: 0.8rem; }
+  .footer-col { display: flex; flex-direction: column; gap: 0.4rem; }
+  .footer-col h4 {
+    font-family: var(--font-mono);
+    font-size: 0.68rem;
+    text-transform: lowercase;
+    letter-spacing: 0.06em;
+    color: #4a6a4a;
+    margin-bottom: 0.25rem;
+  }
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/components/Footer.astro:55]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/components/NavDropdown.astro [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+<div class="nav-dropdown" data-nav-dropdown>
+  <a href={href} class="nav-link nav-dropdown-trigger">
+    {label}
+    <svg class="nav-dropdown-chevron" width="12" height="12" viewBox="0 0 12 12" aria-hidden="true">
+      <path d="M2.5 4.5 6 8l3.5-3.5" fill="none" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" />
+    </svg>
+  </a>
+  {
+    groups.length > 0 && (
+      <div class="nav-dropdown-panel" id={menuId} role="navigation" aria-label={`${label} pages`}>
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/components/NavDropdown.astro:21]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/components/SearchDropdown.astro [hunk 9]
```diff
@@ -0,0 +1,454 @@
...
+    </button>
+  </div>
+
+  <div class="search-menu" id="search-menu" role="dialog" aria-label="Search">
+    <div class="search-input-wrapper">
+      <div id="search-input-container" class="search-input-container"></div>
+      <kbd class="search-close-kbd">esc</kbd>
+    </div>
+    <div class="search-results" id="search-results"></div>
+  </div>
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/components/SearchDropdown.astro]: Fixed: `aria-modal`, `aria-expanded`/`aria-controls` on trigger, Tab focus trap while open, and `searchInitialized` only set after Pagefind loads.

File: apps/site/src/components/SearchDropdown.astro [hunk 10]
```diff
@@ -0,0 +1,454 @@
...
+      })();
+
+      try {
+        await initSearchPromise;
+      } finally {
+        if (!searchInitialized) {
+          initSearchPromise = null;
+        }
+      }
+    }
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/components/SearchDropdown.astro:122]: Fixed: `aria-modal`, `aria-expanded`/`aria-controls` on trigger, Tab focus trap while open, and `searchInitialized` only set after Pagefind loads.

File: apps/site/src/components/ThemeDropdown.astro [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+
+  function onThemePickerDocumentClick(event: MouseEvent): void {
+    const picker = document.getElementById('theme-picker');
+    if (!picker || picker.contains(event.target as Node)) return;
+    picker.classList.remove('is-open');
+    const trigger = document.getElementById('theme-picker-trigger');
+    const panel = document.getElementById('theme-picker-panel');
+    trigger?.setAttribute('aria-expanded', 'false');
+    panel?.setAttribute('hidden', '');
+  }
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/components/ThemeDropdown.astro:79]: Fixed: shared close via WeakMap controller so outside-click/Escape use the same ARIA cleanup as `setOpen(false)`.

File: apps/site/src/components/ui/Card.astro [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+<article
+  class:list={[
+    "card",
+    variant === "featured" && "card-featured",
+    variant === "primary" && "card-primary",
+    feature && "card-feature",
+    className,
+  ]}
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/components/ui/Card.astro:20]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/components/ui/Figure.astro [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+  caption,
+  width,
+  height,
+  loading = "lazy",
+  class: className,
+} = Astro.props;
+---
+
+<figure class:list={["site-figure", className]}>
+  <img src={src} alt={alt} width={width} height={height} loading={loading} />
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/components/ui/Figure.astro:24]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/data/docs-nav.ts [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+  if (inCategory.length === 0) {
+    return `${base}docs/`;
+  }
+
+  return `${base}docs/${inCategory[0]!.id}/`;
+}
+
+export function sectionDocCount(docs: DocNavEntry[], category: DocCategory): number {
+  return docs.filter((d) => d.data.category === category && d.data.sidebar !== false).length;
+}
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/data/docs-nav.ts:98]: Fixed: nav dropdown now respects `sidebar: false` like `sectionDocCount`.

File: apps/site/src/data/sidebar-nav.ts [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+      : [...byGroup.keys()].sort();
+
+  const groups: SidebarNavGroup[] = orderedKeys.map((key) => ({
+    key,
+    label: groupLabels[key] ?? key,
+    docs: (byGroup.get(key) ?? []).sort((a, b) => a.data.order - b.data.order),
+  }));
+
+  return { groups };
+}
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/data/sidebar-nav.ts:139]: Fixed: ungrouped docs fall into `other`; unlisted groups are appended after `groupOrder` instead of being dropped.

File: apps/site/src/lib/rehype-doc-links.mjs [hunk 16]
```diff
@@ -0,0 +1,279 @@
...
+ * applying them to every citation makes documentation difficult to scan.
+ *
+ * @param {string} basePath
+ * @returns {(tree: Root) => void}
+ */
+/**
+ * Turn cross-page prose anchors into plain text so Resources can own SmartLinks.
+ *
+ * @returns {(tree: Root, file: import('vfile').VFile) => void}
+ */
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/lib/rehype-doc-links.mjs:44]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/lib/rehype-doc-links.mjs [hunk 17]
```diff
@@ -0,0 +1,279 @@
...
+
+      const index = siblings.indexOf(node);
+      if (index === -1) {
+        return;
+      }
+
+      siblings[index] = { type: 'text', value: text };
+    });
+  };
+}
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/lib/rehype-doc-links.mjs:85]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/lib/remark-unwrap-cross-page-links.mjs [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+
+      if (!isCrossPageLink(node.url, docId)) {
+        return;
+      }
+
+      parent.children[index] = {
+        type: 'text',
+        value: markdownLinkText(node),
+      };
+    });
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/lib/remark-unwrap-cross-page-links.mjs:30]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/pages/design/docs/filetree.astro [hunk 19]
```diff
@@ -0,0 +1,106 @@
...
+  .keys {
+    margin: 1.25rem 0 0;
+    padding-top: 0.75rem;
+    border-top: 1px dashed #2a4a2a;
+    font-size: 0.65rem;
+    color: #4a6a4a;
+  }
+  .main { padding: 1.75rem 2rem 3rem; min-width: 0; }
+  @media (max-width: 768px) { .layout { grid-template-columns: 1fr; } }
+</style>
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/pages/design/docs/filetree.astro:106]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: apps/site/src/pages/docs/index.astro [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+import { CATEGORY_LABELS, CORE_DOC_CATEGORIES } from '../../data/docs-nav';
+
+const base = import.meta.env.BASE_URL;
+const docs = await getCollection('docs', ({ data }) => !data.draft);
+
+const coreGrouped = CORE_DOC_CATEGORIES.map((key) => ({
+  key,
+  label: CATEGORY_LABELS[key],
+  docs: docs.filter((d) => d.data.category === key).sort((a, b) => a.data.order - b.data.order),
+})).filter((g) => g.docs.length > 0);
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/site/src/pages/docs/index.astro:17]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: scripts/ci/site/fix-markdown-docs.py [hunk 21]
```diff
@@ -0,0 +1,341 @@
...
+        level = len(match.group(1))
+        title = match.group(2).rstrip(":").strip()
+        if previous_level and level > previous_level + 1:
+            level = previous_level + 1
+
+        previous_level = level
+        output.append(f"{'#' * level} {title}")
+
+    return output
+
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply scripts/ci/site/fix-markdown-docs.py:192]: Fixed: `fix_heading_levels` now skips fenced code (same `in_code` pattern as sibling helpers). Also skip blockquote lines in `wrap_prose_lines`.

File: scripts/ci/site/fix-markdown-docs.py [hunk 22]
```diff
@@ -0,0 +1,341 @@
...
+            output.append(line)
+            continue
+
+        indent = line[: len(line) - len(line.lstrip())]
+        wrap_width = max(20, LINE_LENGTH - len(indent))
+        output.extend(
+            f"{indent}{wrapped}" for wrapped in wrap_line(line.strip(), wrap_width)
+        )
+
+    return output
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply scripts/ci/site/fix-markdown-docs.py:315]: Fixed: `fix_heading_levels` now skips fenced code (same `in_code` pattern as sibling helpers). Also skip blockquote lines in `wrap_prose_lines`.

File: scripts/ci/site/migrate-docs-content.py [hunk 23]
```diff
@@ -0,0 +1,370 @@
...
+    route_map: dict[str, str] = {}
+    if DOCS_DEST.exists():
+        for child in DOCS_DEST.iterdir():
+            if child.is_dir():
+                import shutil
+
+                shutil.rmtree(child)
+            else:
+                child.unlink()
+
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply scripts/ci/site/migrate-docs-content.py:264]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: scripts/ci/site/test-python.sh [hunk 24]
```diff
@@ -0,0 +1,28 @@
...
+		uv sync --extra test --frozen
+	else
+		uv sync --extra test --frozen 2>/dev/null || uv sync --extra test
+	fi
+	uv run --extra test pytest tests/scripts/ci -q
+	exit 0
+fi
+
+python3 -m pip install --disable-pip-version-check -q -U "pytest>=8.3"
+python3 -m pytest tests/scripts/ci -q
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply scripts/ci/site/test-python.sh:28]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: scripts/ci/site/test.sh [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+
+cd "${SITE_DIR}"
+
+if ! command -v bun >/dev/null 2>&1; then
+	echo "test.sh: bun is required" >&2
+	exit 1
+fi
+
+bun install --frozen-lockfile
+bun run test
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply scripts/ci/site/test.sh:24]: N/A for merge: trivial/style suggestion; deferred to follow-up polish on the docs site.

File: scripts/ci/testing/stage-python-coverage-html.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply scripts/ci/testing/stage-python-coverage-html.sh]: Fixed: do not pass Cobertura/JSON to `coverage html --data-file`. Prefer `.coverage`, then `htmlcov/`, then render a summary index from `coverage.json` (what the CI artifact actually contains). Also corrected ROOT to `../../..`.
